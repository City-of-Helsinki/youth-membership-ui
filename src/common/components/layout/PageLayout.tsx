import React, { ReactNode } from 'react';

import Header from './header/Header';
import Footer from './footer/Footer';
import PageWrapper from '../wrapper/PageWrapper';
import styles from './PageLayout.module.css';

type Props = {
  children: ReactNode;
};

function PageLayout({ children }: Props) {
  return (
    <PageWrapper>
      <div className={styles.layout}>
        <div className={styles.background} />
        <Header />
        {children}
        <Footer />
      </div>
    </PageWrapper>
  );
}

export default PageLayout;
