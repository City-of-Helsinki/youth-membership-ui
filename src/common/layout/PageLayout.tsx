import React, { ReactNode, useEffect } from 'react';
import classNames from 'classnames';
import { HelmetProvider, Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';
import { useMatomo } from '@datapunt/matomo-tracker-react';

import Header from '../header/Header';
import HostingBox from '../hostingBox/HostingBox';
import Footer from '../footer/Footer';
import styles from './PageLayout.module.css';

type Props = {
  children: ReactNode;
  title?: string;
  background: 'youth' | 'adult';
};

function PageLayout({ children, title = 'appName', background }: Props) {
  const { t, i18n } = useTranslation();
  const { trackPageView } = useMatomo();

  const pageTitle =
    title !== 'appName' ? `${t(title)} - ${t('appName')}` : t('appName');

  useEffect(() => {
    trackPageView({
      documentTitle: pageTitle,
      href: window.location.href,
    });
  });

  return (
    <HelmetProvider>
      <Helmet>
        <html lang={i18n.languages[0]} />
        <title>{t('appName')}</title>
        <meta name="description" content={t('login.helpText')} />
        <meta property="og:title" content={t('appName')} />
        <meta property="og:description" content={t('login.helpText')} />
      </Helmet>
      <div
        className={classNames(
          styles.background,
          background === 'youth'
            ? styles.youthBackground
            : styles.adultBackground
        )}
      >
        <Header />
        <HostingBox className={styles.hostingBox}>{children}</HostingBox>
        <Footer />
      </div>
    </HelmetProvider>
  );
}

export default PageLayout;
