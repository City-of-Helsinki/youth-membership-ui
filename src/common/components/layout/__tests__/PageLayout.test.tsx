import React from 'react';

import { render } from '../../../test/testing-library';
import PageLayout from '../PageLayout';

test('matches snapshot', () => {
  const wrapper = render(
    <PageLayout background="youth">
      <div />
    </PageLayout>
  );

  expect(wrapper.container).toMatchSnapshot();
});
