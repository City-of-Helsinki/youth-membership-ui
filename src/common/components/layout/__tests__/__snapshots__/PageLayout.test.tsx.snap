// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<PageWrapper>
  <div
    className="background"
  >
    <Header />
    <HostingBox
      className="hostingBox"
    >
      <div />
    </HostingBox>
    <Footer />
  </div>
</PageWrapper>
`;
